File: src/main/java/leetcode/medium/restore_ip_addresses/SolutionTest.java
package leetcode.medium.restore_ip_addresses;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class SolutionTest {

    Solution solution = new Solution();

    @Test
    void test1() {
        String s = "25525511135";
        List<String> expected = List.of("255.255.11.135", "255.255.111.35");
        assertThat(solution.restoreIpAddresses(s)).isEqualTo(expected);
    }

    @Test
    void test2() {
        String s = "0000";
        List<String> expected = List.of("0.0.0.0");
        assertThat(solution.restoreIpAddresses(s)).isEqualTo(expected);
    }

    @Test
    void test3() {
        String s = "101023";
        List<String> expected = List.of("1.0.10.23", "1.0.102.3", "10.1.0.23", "10.10.2.3", "101.0.2.3");
        assertThat(solution.restoreIpAddresses(s)).isEqualTo(expected);
    }

}

File: src/main/java/leetcode/medium/reverse_words_in_a_string/SolutionTest.kt
package leetcode.medium.reverse_words_in_a_string

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SolutionTest {

    val solution = Solution()

    @Test
    fun test1() {
        val s = "the sky is blue"
        assertThat(solution.reverseWords(s)).isEqualTo("blue is sky the")
    }

    @Test
    fun test2() {
        val s = "  hello world  "
        assertThat(solution.reverseWords(s)).isEqualTo("world hello")
    }

    @Test
    fun test3() {
        val s = "a good   example"
        assertThat(solution.reverseWords(s)).isEqualTo("example good a")
    }

}